src/main/java/eu/ha3/presencefootsteps/sound/acoustics/ChanceAcoustic.java

```diff
@@ -5,30 +5,31 @@
 import eu.ha3.presencefootsteps.sound.State;
 import eu.ha3.presencefootsteps.sound.player.SoundPlayer;
 import net.minecraft.entity.LivingEntity;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+record ChanceAcoustic(
+        @NotNull Acoustic acoustic,
+        float probability
+) implements Acoustic {
+
+    @Contract(value = "_, _ -> new", pure = true)
+    public static @NotNull Acoustic fromJson(
+            final @NotNull JsonObject json,
+            final @NotNull AcousticsJsonParser context)
+    {
+        final Acoustic acoustic = context.solveAcoustic(json.get("acoustic"));
+        final float probability = json.get("probability").getAsFloat();
 
-class ChanceAcoustic implements Acoustic {
-
-    protected final Acoustic acoustic;
-
-    protected final float probability;
-
-    public ChanceAcoustic(Acoustic acoustic, float probability) {
-        this.acoustic = acoustic;
-        this.probability = probability;
+        return new ChanceAcoustic(acoustic, probability);
     }
 
     @Override
     public void playSound(SoundPlayer player, LivingEntity location, State event, Options inputOptions) {
-        float rand = player.getRNG().nextFloat();
+        final float rand = player.getRNG().nextFloat();
 
         if (rand * 100 <= probability) {
             acoustic.playSound(player, location, event, inputOptions);
         }
     }
-
-    public static Acoustic fromJson(JsonObject json, AcousticsJsonParser context) {
-        Acoustic acoustic = context.solveAcoustic(json.get("acoustic"));
-        float probability = json.get("probability").getAsFloat();
-        return new ChanceAcoustic(acoustic, probability);
-    }
-}
+}
```

Comment:
Newline please

Reply:
Not going to point out each one, just please check through the commit for the others. If there's files without terminating newlines already, it's a style issue on my end that I'll fix as I see them.